client/app/utils/initialize-env.js

```diff
@@ -0,0 +1,12 @@
+import { initialize as initializeI18n } from './i18n';
+import { initialize as initializeRoutes } from './routes';
+
+// This function is a single point where application initialization
+// should happen. Initialization can be for example setting the app
+// in correct state based on railsContext or node environment
+function initializeEnvironment(railsContext, nodeEnv) {
+  initializeI18n(railsContext, nodeEnv);
+  initializeRoutes(railsContext);
+}
```

Comment:
How about instead of passing in the whole railsContext to i18n/routes initializers destructure here and pass in only what's needed. Feels like unnecessary coupling to railsContext now.